wave_controls/src/pid.cpp

```diff
@@ -0,0 +1,51 @@
+#include "wave/controls/pid.hpp"
+
+namespace wave {
+
+PID::PID(void) {
+    this->error_prev = 0.0;
+    this->error_sum = 0.0;
+
+    this->error_p = 0.0;
+    this->error_i = 0.0;
+    this->error_d = 0.0;
+
+    this->k_p = 0.0;
+    this->k_i = 0.0;
+    this->k_d = 0.0;
+}
+
+PID::PID(double k_p, double k_i, double k_d) {
+    this->error_prev = 0.0;
+    this->error_sum = 0.0;
+
+    this->error_p = 0.0;
+    this->error_i = 0.0;
+    this->error_d = 0.0;
+
+    this->k_p = k_p;
+    this->k_i = k_i;
+    this->k_d = k_d;
+}
+
+double PID::update(double setpoint, double actual, double dt) {
+    double error;
```

Comment:
This will require more discussion, as I am not sure the AAA style is a good rule to follow. It is indeed very useful for-range loops, making containers declarations shorter and templating. But the overuse of `auto` will at some point hinder readability when types do matter, given that our team is not at as high level of proficiency at C++ as you are I regard the overuse of AAA style a disadvantage rather than an advantage. Again open to discussion if you have strong opinions about this.